lib/api/core/auth/passportWrapper.js

```diff
@@ -0,0 +1,35 @@
+var
+  q = require('q'),
+  passport = require('passport');
+
+module.exports = function PassportWrapper (kuzzle) {
+
+  this.authenticate = function(request, strategy) {
+    var
+      deferred = q.defer();
+
+    try {
+      if (!passport._strategy(strategy)) {
+        return Promise.reject(new Error('Unknown authentication strategy "' + strategy + '"'));
```

Comment:
Hmm right. I guess the BadRequestError is more appropriate here, as the strategy must be set and mapped to an existing one.